wradlib/io/xarray.py

```diff
@@ -766,19 +766,36 @@ def _preprocess_moment(ds, mom):
     ds = ds.rename({'data': quantity,
                     dims[0]: 'azimuth',
                     dims[1]: 'range'})
+    if mom.parent.decode_coords:
+        coords = mom.parent._get_coords().coords
+        ds = ds.assign_coords(coords)
+        ds = ds.sortby('azimuth')
+        ds = ds.pipe(_reindex_azimuth, mom.parent)
 
     return ds
 
 
 def _reindex_azimuth(ds, sweep, force=False):
+
     dim = list(ds.dims)[0]
     diff = ds[dim].diff(dim)
+
     # this captures missing/additional rays and different angle spacing
     if force | (len(set(diff.values)) > 1):
+        warnings.warn("WRADLIB: missing/additional rays detected "
+                      "=> trying to fix", UserWarning)
+
         # find exact duplicates and remove
         _, idx = np.unique(ds['azimuth'], return_index=True)
         if len(idx) < len(ds['azimuth']):
             ds = ds.isel(azimuth=idx)
+
+            # recompute rtimes if based on nrays
+            sweep._nrays = len(idx)
```

Comment:
Ahh, I see, you are cheating/overriding the nrays/and rtime to be recalculated for the correct amount of rays.